bindings/python/Cargo.toml

```diff
@@ -13,12 +13,12 @@ rayon = "1.3"
 serde = { version = "1.0", features = [ "rc", "derive" ]}
 serde_json = "1.0"
 libc = "0.2"
-env_logger = "0.7.1"
-pyo3 = "0.12"
-numpy = "0.12"
-ndarray = "0.13"
+env_logger = "0.9.0"
+pyo3 = "0.16.2"
```

Comment:
I think the remaining test failures could be resolved by adding

```toml
resolver = "2" # or edition = "2021"

[dev-dependencies]
pyo3 = { version = "0.16", features = ["auto-initialize"]
```

This used to be part of the default features but is not any more since 0.14.

(If Rust 1.51 [which introduced the `resolver = "2"`](https://doc.rust-lang.org/edition-guide/rust-2021/default-cargo-resolver.html#details) option, then the feature can just be added to the normal `[dependencies]` entry.)